### ISSUE TITLE
adding -c flags by default when building go test

### ISSUE THREAD

**Issue by fl0Lec** [goreleaser/goreleaser#4462]
### Is your feature request related to a problem? Please describe.

when building a go test without the `-c` flag like so:
```
 builds:
- command: test
  binary: env.test
  dir: ./internal/builders/golang
  no_main_check: true
  ```

will result in the error: `exit status 1: fork/exec : exec format error`

### Describe the solution you'd like

adding:
```golang
if build.Command == "test" && !slices.Contains(flags, "-c") {
	cmd = append(cmd, "-c")
}
```
in [internal/builders/golang/build.go#L286](https://github.com/goreleaser/goreleaser/blob/8fac823f81b31cebad47eb1dfcdaaa7942d3e390/internal/builders/golang/build.go#L286) would add the -c flag when not present by default

I can come up with a PR but I did not manage to push some code

### Describe alternatives you've considered

clarifying the documentation at [builds.md](https://github.com/goreleaser/goreleaser/blob/8fac823f81b31cebad47eb1dfcdaaa7942d3e390/www/docs/customization/builds.md#L182) to add the requirements to put the `-c` flags

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
hey, yeah, feel free to PR it!
thanks, and sorry for the delayed response.

**Comment by caarlos0** [comment]
closed in #4473
